src/lib/network_pool/transaction_pool.ml

```diff
@@ -3156,4 +3163,744 @@ let%test_module _ =
+      let%bind minor_length = Int.gen_incl 0 sequence_max_length in
+      let%bind prefix_commands =
+        Simple_command.gen_sequence_and_update_ledger ledger
+          ~length:prefix_length
+      in
+      let minor = Simple_ledger.copy ledger in
+      let%bind minor_commands =
+        Simple_command.gen_sequence_and_update_ledger minor ~length:minor_length
+      in
+      let major = Simple_ledger.copy ledger in
+      let%bind major_commands =
+        Simple_command.gen_sequence_and_update_ledger major ~length:major_length
+      in
+      return { prefix_commands; major_commands; minor_commands; minor; major }
+
+    (** Optional Edge Case 1: Limited Account Capacity
+
+        - In major sequence*, a transaction `T` from a specific account
+          decreases its balance by amount `X`.
+        - In minor sequence*, the same account decreases its balance in a
+          similar transaction `T'`, but by an amount much smaller than `X`,
+          followed by several other transactions using the same account.
+        - The prefix ledger* contains just enough funds to process major
+          sequence, with a small surplus.
+        - When applying *minor sequence* without the transaction `T'` (of the
+          same nonce as the large-amount transaction `T` in major sequence), the
+          sequence becomes partially applicable, forcing the mempool logic to
+          drop some transactions at the end of *minor sequence*.
+    *)
+    let gen_updated_branches_for_limited_capacity
+        { prefix_commands; major_commands; minor_commands; minor; major } =
+      let open Quickcheck.Generator.Let_syntax in
+      (*find account in major and minor branches with the same nonces and similar balances (less than 100k mina diff)*)
+      let%bind target_account_idx, target_account =
+        Simple_ledger.get_random_unsealed major
+      in
+      let initial_nonce = Simple_account.nonce target_account in
+      (* find receiver which is not our selected account*)
+      let%bind receiver_idx =
+        test_keys
+        |> Array.filter_mapi ~f:(fun i _ ->
+               if Int.equal i (Simple_account.key_idx target_account) then None
+               else Some i )
+        |> Quickcheck_lib.of_array
+      in
+      let%bind major_sequence_length = Int.gen_incl 2 10 in
+      let%bind minor_sequence_length =
+        let%map minor_sequence_length = Int.gen_incl 2 4 in
+        minor_sequence_length + major_sequence_length + initial_nonce
+      in
+      let initial_balance = Simple_account.balance target_account in
+      let half_initial_balance = Simple_account.balance target_account / 2 in
+      let gen_sequence_and_update_account ledger len =
+        let account = ref (Simple_ledger.get ledger target_account_idx) in
+        let%map sequence =
+          Quickcheck_lib.init_gen_array len ~f:(fun _ ->
+              let%bind amount =
+                Int.gen_incl 5_000_000_000_000_000 (half_initial_balance / len)
```

Comment:
What if `(half_initial_balance / len) < 5_000_000_000_000_000`? will it fail with an exception?

Reply:
So, I just checked that initial balance is always 900mln. I still see little benefit in hardcoding the `5mln` amount here

Reply:
hm, so what should be the minimal amount ? should it be related to half_initial_balance?

Reply:
yeah, let's make it proportional to `half_initial_balance/len`